### PR TITLE
Extract `ComposeContainer`

### PR DESCRIPTION
**Pull request by MatkovIvan**
Another part of #939

### CODE REVIEW

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    override fun windowGainedFocus(event: WindowEvent) = onChangeWindowFocus()
+    override fun windowLostFocus(event: WindowEvent) = onChangeWindowFocus()
+
+    private fun onChangeWindowFocus() {
+        windowContext.setWindowFocused(window?.isFocused ?: false)
+        bridge.onChangeWindowFocus()
+    }
+
+    private fun onChangeWindowBounds() {
+        val component = window ?: container
```
Comment (igordmn) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: Can we just use `container`, or use the passed `setBounds`?

Checks likes this add inderection - a child (`ComposeContainer`) checks the parent (`ComposePanel`/`ComposeWindowPanel`) type. Better to the parent to tell what child should do.

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    init {
+        setWindow(window)
+    }
+
+    fun dispose() {
+        bridge.dispose()
+    }
+
+    override fun componentResized(e: ComponentEvent?) = onChangeWindowBounds()
```
Comment (igordmn) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: Do we need to subscribe to componentResized/componentMoved, if we already have `setBounds`?

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt [hunk 3]
```diff
@@ -225,7 +235,7 @@ internal class ComposeBridge(
             }
         })
 
-        component.addFocusListener(object : FocusListener {
+        contentComponent.addFocusListener(object : FocusListener {
```
Comment (igordmn) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt:237]: Let's not forget to unsubscribe (in a separate PR, or carefully in this), as ComposeBridge is now uncoupled from the component creation.

Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt:237]: In a separate one

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt (hunk 2 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: Yes. `setBounds` is about changing container, not a window

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: No. It was added to update layers about window size, not a container

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt (hunk 2 above)
Comment (igordmn) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: In this case, see my other comment about inderection. We can subscribe to componentResized inside `ComposeWindowPanel`

**Review by igordmn — APPROVED** [review]

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.kt]: Removed it for now, but it will be needed later

**Pull request merged**